src/files.ts

```diff
@@ -0,0 +1,21 @@
+import { appendFile, existsSync, writeFile } from 'fs';
+import { resolve as resolvePath } from 'path';
+
+const FILE_NAME = resolvePath(__dirname, '../src/assets/custom-styles.css');
+
+export function write(data: any) {
+  if (existsSync(FILE_NAME)) {
+    return new Promise((resolve, reject) => {
+      appendFile(FILE_NAME, data, (err: any) => {
+        if (err) reject(err);
+        resolve(data);
+      });
+    });
+  }
+  return new Promise((resolve, reject) => {
```

Comment:
The complexity is high with this one, probably going to refactor moving forward, but pull request to me makes it seems like this code is going into master.